src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,50 @@
+package duke.task;
+import duke.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+
+    protected String deadline;
+    protected String date = "";
+
+    public Deadline(String description, String deadline) throws DukeException {
+        super(description);
+
+        try {
+            LocalDate localDate = LocalDate.parse(deadline);
+            this.date = localDate.format(DateTimeFormatter.ofPattern("MMM dd yyyy"));
+        } catch (DateTimeParseException e) {
+            throw new DukeException("duke.task.Deadline should be in a yyyy-mm-dd format.");
+        }
+
+        if (description.isEmpty() || description == "" || description == " ") {
+            throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+        } else {
+            this.description = description.substring(1);
+        }
+
+        if (deadline.isEmpty() || deadline == "" || deadline == " ") {
+            throw new DukeException("☹ OOPS!!! The deadline of this task must be indicated.");
```

Comment:
As before, these repeated checks in the if statements should not be necessary after using `input.split(" ", 2)` in `Parser.java`.